raiden-ts/src/raiden.ts

```diff
@@ -319,7 +320,7 @@ export class Raiden {
       | RaidenState
       | { storage: Storage; state?: RaidenState | unknown }
       | unknown,
-    contracts?: ContractsInfo,
+    contractsOrUserDepositAddress?: ContractsInfo | string,
```

Comment:
Oh I hate such code so damn much. Actually getting goose bumps while reading it. But since TypeScript doesn't support function overloading I guess we have to do it like this to maintain backwards compatibility.

Reply:
Yes. It actually do support function overloads, but only at interface level, so we could have the declared overloads with the shorter variables, but then we'd need an implementation with the full name like here. I'm not a big fan of it either, but since it's used only on this function and doesn't leak on the rest of the implementation, I'm fine with it also to maintain backwards compatibility.